File: frontend/src/api/dto/loginRequest.ts
/**
 * Password login request
 */
export interface LoginRequest {
  /**
   * Email address
   * @maxLength 250
   * @pattern \S
   */
  email: string
  /**
   * Password
   * @maxLength 100
   * @pattern \S
   */
  password: string
}

File: backend/src/main/java/fr/pedalons/dto/auth/request/LoginRequest.java
package fr.pedalons.dto.auth.request;

import fr.pedalons.dto.validation.ValidateSchema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Password login request")
@ValidateSchema
@Builder
public record LoginRequest(
    @NotBlank
        @Email
        @Size(max = 250)
        @Schema(description = "Email address", examples = "user@example.com")
        String email,
    @NotBlank @Size(max = 100) @Schema(description = "Password") String password) {}
